.editorconfig

```diff
@@ -148,6 +148,21 @@ dotnet_diagnostic.RS2008.severity = none
 dotnet_diagnostic.IDE0073.severity = warning
 file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.\nSee the LICENSE file in the project root for more information.
 
+# IDE0035: Remove unreachable code
+dotnet_diagnostic.IDE0035.severity = warning
+
+# IDE0036: Order modifiers
+dotnet_diagnostic.IDE0036.severity = warning
+
+# IDE0040: Add accessibility modifiers
+dotnet_diagnostic.IDE0040.severity = warning
+
+# IDE0043: Format string contains invalid placeholder
+dotnet_diagnostic.IDE0043.severity = warning
+
+# IDE0044: Make field readonly
+dotnet_diagnostic.IDE0044.severity = warning
+
```

Comment:
Do you want to enforce any of these? There seem to be violations of them currently.

Reply:
Yes, absolutely. Based on our offline conversation with @jaredpar and @sharwell we want to uniformly enforce code style rules across the Roslyn repo (as much as possible, some code styles might have different preferences between compiler and IDE parts, but that is fine).

I think your PR represents a great step towards our eventual goal of moving all/majority of the code style enforcement entries from the below IDE specific section to the repo wide section.